### PR TITLE
Fix some issues connecting to firefox

### PR DESCRIPTION
**Pull request by jasonLaster**
This fixes two issues connecting to firefox:
1.  refreshing several times https://github.com/devtools-html/devtools-core/issues/199
2. re-opening firefox

We also do some cleanup
1. add react 15.3.2 to core so that the launchpad can be launched by itself
2. add a simple connected component to the launchpad to test the connection
3. refactor the dev-server so that it is in src and can be modularized and linted. We'll likely want to move some src code for launchpad into a `client` dir

### CODE REVIEW

File: packages/devtools-sham-modules/sham/chrome.js [hunk 1]
```diff
@@ -19,25 +19,23 @@ var ourServices = {
 
 module.exports = {
   Cc: name => {
-    if(typeof console !== "undefined") {
-      console.log('Cc sham for', name);
+    if (typeof console !== "undefined") {
```
Comment (juliandescottes) [inline packages/devtools-sham-modules/sham/chrome.js]: why the empty if ?

File: packages/devtools-sham-modules/sham/chrome.js [hunk 2]
```diff
@@ -19,25 +19,23 @@ var ourServices = {
...
     }
     return {
       getService: (name) => ourServices[name],
       createInstance: (iface) => ourServices[iface],
     };
   },
   CC: (name, iface, method) => {
-    if(typeof console !== "undefined") {
-      console.log('CC sham for', name, iface, method);
+    if (typeof console !== "undefined") {
```
Comment (juliandescottes) [inline packages/devtools-sham-modules/sham/chrome.js]: same comment

File: packages/devtools-launchpad/src/index.js [hunk 3]
```diff
@@ -159,7 +164,7 @@ async function bootstrap(React, ReactDOM, App, appActions, appStore) {
 
   const { store, actions, LaunchpadApp } = await initApp();
   renderRoot(React, ReactDOM, LaunchpadApp, store);
-  await getTabs(actions);
+  await connectClients(actions);
```
Comment (juliandescottes) [inline packages/devtools-launchpad/src/index.js]: If I read this correctly, every 3 seconds we empty the tab list and fetch the firefox tabs.
Isn't it a problem when debugging Chrome/Node?

File: packages/devtools-launchpad/src/index.js [hunk 4]
```diff
@@ -144,6 +143,12 @@ async function getTabs(actions) {
   actions.newTabs(firefoxTabs);
 }
 
+async function getTabs(actions) {
+  actions.clearTabs();
```
Comment (juliandescottes) [inline packages/devtools-launchpad/src/index.js]: This won't pickup newly started firefox instances. If you do:
- yarn start in launchpad
- go to localhost:8000
- click on "Launch Firefox"
=> you now have to reload the page to see the tab from the new firefox instance
(previously it was automatically picked up)

If the connection was not established, maybe we should still try to do connectClients()?

File: packages/devtools-launchpad/src/index.js (hunk 3 above)
Comment (jasonLaster) [inline packages/devtools-launchpad/src/index.js]: hmm... i'll clean that up

File: packages/devtools-launchpad/src/index.js (hunk 4 above)
Comment (jasonLaster) [inline packages/devtools-launchpad/src/index.js]: We do it there too. This is here to catch new tabs.

Comment (juliandescottes) [inline packages/devtools-launchpad/src/index.js]: My point is that getTabs is not trying to connect to new firefox clients. So, as I said, this forces you to refresh your browser after clicking on "Launch firefox".

**Comment by jasonLaster** [comment]
Hmm. I believe it works though. I'll investigate why :)

**Pull request merged**